src/surface.h

```diff
@@ -0,0 +1,427 @@
+#ifndef SURFACE_H
+#define SURFACE_H
+
+#include <map>
+#include <limits>  // For numeric_limits
+#include <string>
+
+#include "hdf5.h"
+#include "pugixml/pugixml.hpp"
+
+
+namespace openmc {
+
+//==============================================================================
+// Module constants
+//==============================================================================
+
+extern "C" const int BC_TRANSMIT {0};
+extern "C" const int BC_VACUUM {1};
+extern "C" const int BC_REFLECT {2};
+extern "C" const int BC_PERIODIC {3};
+
+//==============================================================================
+// Constants that should eventually be moved out of this file
+//==============================================================================
+
+extern "C" double FP_COINCIDENT;
+constexpr double INFTY{std::numeric_limits<double>::max()};
+constexpr int C_NONE {-1};
+
+//==============================================================================
+// Global variables
+//==============================================================================
+
+// Braces force n_surfaces to be defined here, not just declared.
+extern "C" {int32_t n_surfaces {0};}
+
+class Surface;
+Surface **surfaces_c;
+
+std::map<int, int> surface_dict;
+
+//==============================================================================
+//! Coordinates for an axis-aligned cube that bounds a geometric object.
+//==============================================================================
+
+struct BoundingBox
+{
+  double xmin;
+  double xmax;
+  double ymin;
+  double ymax;
+  double zmin;
+  double zmax;
+};
+
+//==============================================================================
+//! A geometry primitive used to define regions of 3D space.
+//==============================================================================
+
+class Surface
+{
+public:
+  int id;                    //!< Unique ID
+  //int neighbor_pos[],        //!< List of cells on positive side
+  //    neighbor_neg[];        //!< List of cells on negative side
+  int bc;                    //!< Boundary condition
+  std::string name{""};      //!< User-defined name
+
+  Surface(pugi::xml_node surf_node);
```

Comment:
Declare this constructor `explicit` (as well as the single-argument constructors for all derived classes).